controllers/machinehealthcheck_controller.go

```diff
@@ -333,20 +330,20 @@ func (r *MachineHealthCheckReconciler) watchClusterNodes(ctx context.Context, c
 	return r.createClusterNodeInformer(ctx, c, key)
 }
 
-func (r *MachineHealthCheckReconciler) getClusterNodeInformer(key client.ObjectKey) (cache.Informer, bool) {
-	r.clusterNodeInformersLock.RLock()
-	defer r.clusterNodeInformersLock.RUnlock()
+func (r *MachineHealthCheckReconciler) getClusterNodeInformer(key client.ObjectKey) (cache.Cache, bool) {
```

Comment:
```suggestion
func (r *MachineHealthCheckReconciler) getClusterNodeCache(key client.ObjectKey) (cache.Cache, bool) {
```

Reply:
Good spot! This is why I shouldn't rebase code on a Monday morning 😪 ☕️ 😂 Thanks @detiber